aten/src/ATen/native/cuda/Blas.cpp

```diff
@@ -496,12 +496,6 @@ inline void dot_check(const Tensor& self, const Tensor& other) {
       " and ",
       other.numel(),
       " elements respectively");
-  TORCH_CHECK(
-      self.device() == other.device(),
-      "expected all tensors to be on the same device. Found: ",
-      self.device(),
-      ", ",
-      other.device());
```

Comment:
@peterbell10 removing this as it's unreachable. We currently check this in the wrapper and throw
```
Expected all tensors to be on the same device, but found at least two devices, cpu and cuda:0! (when checking argument for argument tensor in method wrapper_CUDA__dot
```